### PR TITLE
Fix handling of timestamps when ingesting from CSV

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/chronify/csv_io.py [hunk 1]
```diff
@@ -5,15 +5,28 @@
...
-    expr = ",".join(rel.columns)
+    time_config = schema.time_config
+    exprs = []
+    for i, column in enumerate(rel.columns):
+        expr = column
+        if isinstance(time_config, DatetimeRange) and column == time_config.time_column:
+            time_type = rel.types[i]
+            if time_type == duckdb.typing.TIMESTAMP and time_config.start.tzinfo is not None:  # type: ignore
+                expr = f"timezone('{time_config.start.tzinfo.key}', {column}) AS {column}"  # type: ignore
+        exprs.append(expr)
```
Comment (daniel-thom) [inline src/chronify/csv_io.py:29]: This was the main problem. Now we are applying time zones.

File: src/chronify/store.py [hunk 2]
```diff
@@ -132,16 +132,28 @@ def ingest_from_csv(
...
+                dst_schema.name,
+                self._metadata,
+                *[Column(x, y) for x, y in zip(rel.columns, dtypes)],
+            )
             table.create(self._engine)
 
         with self._engine.begin() as conn:
             write_database(rel.to_df(), conn, dst_schema)
+            try:
+                check_timestamps(conn, table, dst_schema)
```
Comment (daniel-thom) [inline src/chronify/store.py:150]: Previous behavior was incorrect. We need to perform checks on the final table and rollback on error.

File: src/chronify/time_configs.py [hunk 3]
```diff
@@ -183,8 +183,8 @@ def is_time_zone_naive(self) -> bool:
         """Return True if the timestamps in the range do not have time zones."""
         return self.start.tzinfo is None
 
-    def list_timestamps_from_dataframe(self, df: pd.DataFrame) -> list[datetime]:
-        return df[self.time_column].to_list()
+    def list_distinct_timestamps_from_dataframe(self, df: pd.DataFrame) -> list[datetime]:
+        return sorted(df[self.time_column].unique())
```
Comment (daniel-thom) [inline src/chronify/time_configs.py:187]: Previous behavior was incorrect.

File: src/chronify/time_series_checker.py [hunk 4]
```diff
@@ -1,79 +1,79 @@
...
-    @staticmethod
-    def _run_timestamp_checks_on_tmp_table(
-        schema: TableSchema, conn: Connection, table_name: str
-    ) -> None:
-        id_cols = ",".join(schema.time_array_id_columns)
-        filters = [f"{x} IS NOT NULL" for x in schema.time_config.list_time_columns()]
+    def _run_timestamp_checks_on_tmp_table(self, table_name: str) -> None:
+        id_cols = ",".join(self._schema.time_array_id_columns)
+        filters = [f"{x} IS NOT NULL" for x in self._schema.time_config.list_time_columns()]
         where_clause = "AND ".join(filters)
```
Comment (lixiliu) [inline src/chronify/time_series_checker.py:54]: This needs to be " AND ", space in front of and behind.

File: src/chronify/csv_io.py (hunk 1 above)
Comment (lixiliu) [reply src/chronify/csv_io.py:29]: It seems like we may not need any of the timezone enums we created in time.py since we can use tzinfo directly.

Comment (daniel-thom) [reply src/chronify/csv_io.py:29]: That might be the case. We might need them if we accept user input definitions as command line arguments or in configuration files.

File: src/chronify/time_series_checker.py (hunk 4 above)
Comment (daniel-thom) [reply src/chronify/time_series_checker.py]: Thanks. Fixed.

**Pull request merged**